cabal-install/Distribution/Client/SetupWrapper.hs

```diff
@@ -389,26 +389,33 @@ externalSetupMethod verbosity options pkg bt mkargs = do
       Nothing    -> getInstalledPackages verbosity
                     comp (usePackageDB options') conf
 
-  cabalLibVersionToUse :: IO (Version, (Maybe ComponentId)
+  cabalLibVersionToUse :: IO (Version, Maybe ComponentId
                              ,SetupScriptOptions)
   cabalLibVersionToUse =
-    case useCabalSpecVersion options of
-      Just version -> do
+    case find (hasCabal . snd) (useDependencies options) of
```

Comment:
@grayjay I think the comment refers to the fact that if `useDependenciesExclusive` is set, then we should just use those dependencies exclusively. If you look at the code, in both places we set `useDependencies`, we also set `useCabalVersion`.